src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEParameterSymbol.cs

```diff
@@ -305,17 +305,26 @@ private enum IsParamsValues : byte
                     }
                 }
 
-                var typeSymbol = DynamicTypeDecoder.TransformType(typeWithAnnotations.Type, countOfCustomModifiers, handle, moduleSymbol, refKind);
-                typeSymbol = NativeIntegerTypeDecoder.TransformType(typeSymbol, handle, moduleSymbol, containingSymbol.ContainingType);
-                typeWithAnnotations = typeWithAnnotations.WithTypeAndModifiers(typeSymbol, typeWithAnnotations.CustomModifiers);
-                // Decode nullable before tuple types to avoid converting between
-                // NamedTypeSymbol and TupleTypeSymbol unnecessarily.
-
-                // The containing type is passed to NullableTypeDecoder.TransformType to determine access
-                // for property parameters because the property does not have explicit accessibility in metadata.
-                var accessSymbol = containingSymbol.Kind == SymbolKind.Property ? containingSymbol.ContainingSymbol : containingSymbol;
-                typeWithAnnotations = NullableTypeDecoder.TransformType(typeWithAnnotations, handle, moduleSymbol, accessSymbol: accessSymbol, nullableContext: nullableContext);
-                typeWithAnnotations = TupleTypeDecoder.DecodeTupleTypesIfApplicable(typeWithAnnotations, handle, moduleSymbol);
+                if (moduleSymbol.TryDecodeExtensionErasureAttribute(handle, (PENamedTypeSymbol)containingSymbol.ContainingType, containingSymbol as PEMethodSymbol) is { } typeWithExtensions)
+                {
+                    typeWithAnnotations = TypeWithAnnotations.Create(typeWithExtensions);
```

Comment:
>typeWithAnnotations = TypeWithAnnotations.Create(typeWithExtensions) [](http://example.com/codeflow?start=20&length=68)

Should we drop the attribute from the symbol?

Reply:
That would make sense. I think we should keep it in the PEParameterSymbol which represents what is actually in metadata, but remove it from PEExtensionInstanceMethodParameterSymbol.
Would that work?

Reply:
Why would PEExtensionInstanceMethodParameterSymbol be special here? We are actually removing a bunch of attributes that we synthesize